pkg/stockings/fingopack.go

```diff
@@ -0,0 +1,159 @@
+package stockings
+
+import (
+	"time"
+
+	"github.com/piquette/finance-go"
+	"github.com/piquette/finance-go/chart"
+	"github.com/piquette/finance-go/datetime"
+	"github.com/piquette/finance-go/quote"
+
+	"github.com/shopspring/decimal"
+)
+
+// FingoPack will be used to determine which interface we are using
+type FingoPack struct {
+}
+
+// fingoHistoricalStocks is a list of struct ChartBar which is determined by finance-go
+type piqHistoricalStocks []finance.ChartBar
+
+func (piq FingoPack) GetCurrentPrice(ticker string) (*Stock, error) {
+
+	// simple call to get the quote of a stock in the finance.quote struct
+	quote, err := quote.Get(ticker)
+	if err != nil {
+		return nil, err
+	}
+
+	// need to convert our types to match our universal Stock struct - should this be broken out into a seperate function?
+	// some of the types are already correct, but listed them all out for clarity
+	// finance.quote has a HUGE variety of info in it that we can pull
+	symbol := quote.Symbol
+	name := quote.ShortName
+	currentPrice := decimal.NewFromFloat(quote.RegularMarketPrice)
+	priceChange := float32(quote.RegularMarketChange)
+	percentChange := float32(quote.RegularMarketChangePercent)
+
+	return &Stock{symbol, name, currentPrice, priceChange, percentChange}, nil
+
+}
+
+// GetHistoricalPrice is hacky because Fingo only has a function that returns quotes in a time span
+// This function takes in the desired date as the end date, then sets the start date as one month prior
+func (piq FingoPack) GetHistoricalPrice(ticker string, date time.Time) (*HistoricalStock, error) {
+
+	quoteAtDate := new(finance.ChartBar)
+
+	// converting the time.Time struct into usuable integers to be fed into Fingo
+	year := date.Year()
+	month := int(date.Month())
+	day := date.Day()
+
+	// converting into our accepted Fingo form
+	endDate := datetime.Datetime{
```

Comment:
so I looked into `github.com/piquette/finance-go/datetime`, they have a `datetime.New(*time.Time)` function which should be useful for us. golang's `time` package is really nice, and it has builtin time delta functions for you. So instead of having to specifically check for stuff like whether month is 1, you can do something like:
```
endDate := datetime.New(date)
start := date.Sub(date.AddDate(0, -1, 0))
startDate := datetime.New(start)
```
```

Reply:
also, do we actually have to have a window of a month? Is that their smallest window?

Reply:
updated based on your comments and reduced window to 5 days. giving ourselves some cushioning in case there's some apocalyptic scenario